service/src/main/java/uk/gov/hmcts/reform/fpl/model/common/C2DocumentBundle.java

```diff
@@ -29,4 +32,58 @@
     public String toLabel(int index) {
         return format("Application %d: %s", index, uploadedDateTime);
     }
+
+    @JsonIgnore
+    public String getAllC2DocumentFileNames() {
+        String c2Filename = "";
+
+        if (document != null) {
+            c2Filename = document.getFilename();
+        }
+
+        String stringBuilder = c2Filename + "\n" + getSupportingEvidenceFileNames();
+        return stringBuilder.trim();
+    }
+
+    @JsonIgnore
+    public List<Element<DocumentReference>> getAllC2DocumentReferences() {
+        List<Element<DocumentReference>> documentReferences = new ArrayList<>();
+
+        if (document != null) {
+            documentReferences.add(element(document));
+        }
+
+        documentReferences.addAll(getSupportingEvidenceBundleReferences());
+
+        return documentReferences;
+    }
+
+    @JsonIgnore
+    private String getSupportingEvidenceFileNames() {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (supportingEvidenceBundle != null) {
```

Comment:
that could be rewritten using immutability rather than a stringbuffer. Look stream.map().collect(Collector.joining())